### PR TITLE
Fix: builder shadow var 2

### PR DESCRIPTION
**Pull request by samijaber**

### CODE REVIEW

File: packages/core/src/__tests__/data/blocks/index-in-for.raw.tsx [hunk 1]
```diff
@@ -10,12 +10,14 @@ export default function MyComponent(props) {
     <div>
       {todos.map((todo) => (
         <div>
-          {state.todos.map((todo, index) => (
+          {state.todos.map((todo) => (
             <input
               type="checkbox"
               checked={todo.completed}
+              key={todo.id + '-' + index}
```
Comment (manucorporat) [inline packages/core/src/__tests__/data/blocks/index-in-for.raw.tsx:17]: index is undefined, no?

Comment (samijaber) [reply packages/core/src/__tests__/data/blocks/index-in-for.raw.tsx:17]: ah good point. i should have kept `index` in the map for this

File: packages/core/src/__tests__/data/blocks/index-in-for.raw.tsx [hunk 2]
```diff
@@ -10,12 +10,14 @@ export default function MyComponent(props) {
     <div>
       {todos.map((todo) => (
         <div>
-          {state.todos.map((todo, index) => (
+          {state.todos.map((todo) => (
```
Comment (samijaber) [inline packages/core/src/__tests__/data/blocks/index-in-for.raw.tsx:13]:
```suggestion
          {state.todos.map((todo, index) => (
```

File: packages/core/src/__tests__/__snapshots__/builder.test.ts.snap [hunk 3]
```diff
@@ -1363,8 +1363,10 @@ export default function MyComponent(props) {
                 <input
                   type=\\"checkbox\\"
                   checked={todo.completed}
+                  key={todo.id + \\"-\\" + index}
+                  value={index}
                   foo={{
-                    bar: 1,
+                    bar: 1 + state.$index,
```
Comment (samijaber) [inline packages/core/src/__tests__/__snapshots__/builder.test.ts.snap:1369]: hmm @manucorporat it's weird that the `key` and `value` here are not getting updated to `state.$index`

I think that's a bug no?

**Pull request merged**